src/test/test-fs-util.c

```diff
@@ -88,6 +90,34 @@ static void test_chase_symlinks(void) {
         assert_se(r >= 0);
         assert_se(path_equal(result, temp));
 
+        /* Paths that would "escape" outside of the "root" */
+
+        p = strjoina(temp, "/6dots");
+        assert_se(symlink("../../..", p) >= 0);
+
+        result = mfree(result);
+        r = chase_symlinks(p, temp, &result);
```

Comment:
I think that if the symlink would go "above" root, it should just stay at root. I.e. "../../../usr" should be the same as "../../usr", if we are not too deep. My point was that doing this is perfectly legal in a real file system (so for example, you can do ../../usr/lib64 from both /lib and /usr/lib and it'll "just work"), so chase_symlinks should also allow that.